api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionsUriHelper.java

```diff
@@ -10,13 +12,17 @@
 public class SubscriptionsUriHelper {
 
     public static PaginationLinks.Link createSubscriptionListLink(
-            final Optional<String> owningApplication, final Set<String> eventTypes, final int offset,
+            final Optional<String> owningApplication, final Set<String> eventTypes,
+            final Optional<AuthorizationAttribute> reader, final int offset,
             final Optional<SubscriptionTokenLister.Token> token, final int limit, final boolean showStatus) {
 
         final UriComponentsBuilder urlBuilder = UriComponentsBuilder.fromPath("/subscriptions");
         if (!eventTypes.isEmpty()) {
             urlBuilder.queryParam("event_type", eventTypes.toArray());
         }
+        reader.ifPresent(_reader -> {
+            urlBuilder.queryParam("reader", AuthorizationAttributeQueryParser.getQuery(_reader));
```

Comment:
and again, that is a good place where model -> view mapping would be good, so you do not need to integrate static method into AuthorizationAttributeQueryParser

Reply:
Can you share an example? I decided to encapsulate this on the parser class so the logic if parsing is kept in one place only.

Reply:
static method looks unnatural in that class. maybe there is the same @InitBinder approach, but to map it back.

Reply:
@InitBinder does not work here unfortuantely.